dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/util/RelationshipKeySupport.java

```diff
@@ -89,35 +64,11 @@ public static RelationshipKey.RelationshipItemKey getRelationshipItemKey(
         throw new IllegalStateException( "Unable to determine uid for relationship item" );
     }
 
-    public static RelationshipKey.RelationshipItemKey getRelationshipItemKey( RelationshipItem relationshipItem )
-    {
-        if ( Objects.nonNull( relationshipItem ) )
-        {
-            return RelationshipKey.RelationshipItemKey.builder()
-                .trackedEntity( getUidOrEmptyString( relationshipItem.getTrackedEntityInstance() ) )
-                .enrollment( getUidOrEmptyString( relationshipItem.getProgramInstance() ) )
-                .event( getUidOrEmptyString( relationshipItem.getProgramStageInstance() ) )
-                .build();
-        }
-        throw new IllegalStateException( "Unable to determine uid for relationship item" );
-    }
-
-    private static String getUidOrEmptyString( BaseIdentifiableObject baseIdentifiableObject )
-    {
-        return Objects.isNull( baseIdentifiableObject ) ? ""
-            : StringUtils.trimToEmpty( baseIdentifiableObject.getUid() );
-    }
-
-    public static boolean isRelationshipPreheatKey( String s )
+    public static RelationshipKey getRelationshipKey( Relationship relationship, String relationshipType )
```

Comment:
I am not a fan of utils 😋 

IMHO these are nicer factories for RelationshipKey/RelationshipKey.RelationshipItemKey
Since they depend on our tracker domain I agree that they should not live in `org.hisp.dhis.relationship.`.

In what packages is this class used? If its only used in preheat I think preheat is a good place.

Reply:
It shouldn't be in Preheat package as it is also used in the converter.
They could live in tracker domain objects but now that we are passing the RelationshipType to create the key it seems forced have them there.

`getRelationshipItemKey` could stay in `RelationshipItem` object as it doesn't depend on anything else